server/controllers/batches.ts

```diff
@@ -656,31 +662,51 @@ export const payStartBonus = async (users, batch) => {
 const generateTeams = (roundGen, users, roundNumber, oldNicks) => {
   let teams = roundGen[roundNumber - 1].teams.map((team, index) => {
     let teamAnimals = [],
-      teamUsers = [];
+      teamUsers = [],
+      teamAdj = [],
+      prevAnimals = [];
+
+    // don't use animals on any of the users' real nicknames
     team.users.forEach((user) => {
-      let partner = users[user].realNick.slice();
-      for (let i = 0; i < partner.length; i++) {
-        if (partner[i] === partner[i].toUpperCase()) {
-          index = i;
-          break;
-        }
-      }
-      teamAnimals.push(partner.slice(0, index));
```

Comment:
Below: Logic for setting unique real names, for every new round, subtract all animals used in user real names and previously seen nick names, if this results in less new names than required in the round, then we use the original full list and just avoid using the same two animals in one round

Reply:
Sounds good!